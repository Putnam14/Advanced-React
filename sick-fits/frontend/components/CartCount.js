import React from 'react'
import PropTypes from 'prop-types'
import { TransitionGroup, CSSTransition } from 'react-transition-group'
import styled from 'styled-components'

const AnimationStyles = styled.span`
  position: relative;
  .count {
    display: block;
    position: relative;
    transition: all 0.4s;
    backface-visibility: hidden;
  }
  /* Initial state of the entered dot */
  /* Rotate from on back to its front */
  .count-enter {
    transform: rotateX(0.5turn); /* Turn is 360 degrees */
  }
  .count-enter-active {
    transform: rotateX(0);
  }
  /* Exit animation */
  .count-exit {
    top: 0;
    position: absolute;
    transform: rotateX(0); /* Turn is 360 degrees */
  }
  .count-exit-active {
    top: 0;
    position: absolute;
    transform: rotateX(-0.5turn);
  }
`

const Dot = styled.div`
  background: ${props => props.theme.red};
  color: white;
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  min-width: 3rem;
  margin-left: 1rem;
  font-weight: 100;
  /* Will make it so that fat numbers don't expand the div */
  font-feature-setting: 'tnum';
  font-variant-numeric: tabular-nums;
`

const CartCount = ({ count }) => (
  <AnimationStyles>
    <TransitionGroup>
      <CSSTransition
        unmountOnExit
        className="count"
        classNames="count"
        key={count}
        timeout={{ enter: 400, exit: 400 }}
      >
        <Dot>{count}</Dot>
      </CSSTransition>
    </TransitionGroup>
  </AnimationStyles>
)

export default CartCount
