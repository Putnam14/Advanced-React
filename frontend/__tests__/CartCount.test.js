import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CartCount from '../components/CartCount';

describe('<CartCount></CartCount>', () => {
  it('renders okay', () => {
    shallow(<CartCount count="10" />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<CartCount count="10" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('updates via props', () => {
    const wrapper = mount(<CartCount count="50" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
