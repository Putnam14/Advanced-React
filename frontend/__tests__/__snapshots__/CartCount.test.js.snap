// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount></CartCount> matches snapshot 1`] = `
<styled.span>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <styled.div>
        10
      </styled.div>
    </CSSTransition>
  </TransitionGroup>
</styled.span>
`;

exports[`<CartCount></CartCount> updates via props 1`] = `
<CartCount
  count="50"
>
  <styled.span>
    <span
      className="sc-bwzfXH fWZfOZ"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <div>
          <CSSTransition
            className="count"
            classNames="count"
            in={true}
            key=".$50"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <styled.div
                className="count"
              >
                <div
                  className="count sc-bdVaJa ldHGxH"
                >
                  50
                </div>
              </styled.div>
            </Transition>
          </CSSTransition>
        </div>
      </TransitionGroup>
    </span>
  </styled.span>
</CartCount>
`;

exports[`<CartCount></CartCount> updates via props 2`] = `
<CartCount
  count={10}
>
  <styled.span>
    <span
      className="sc-bwzfXH fWZfOZ"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <div>
          <CSSTransition
            className="count"
            classNames="count"
            in={true}
            key=".$10"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <styled.div
                className="count"
              >
                <div
                  className="count sc-bdVaJa ldHGxH"
                >
                  10
                </div>
              </styled.div>
            </Transition>
          </CSSTransition>
          <CSSTransition
            className="count"
            classNames="count"
            in={false}
            key=".$50"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <styled.div
                className="count"
              >
                <div
                  className="count sc-bdVaJa ldHGxH"
                >
                  50
                </div>
              </styled.div>
            </Transition>
          </CSSTransition>
        </div>
      </TransitionGroup>
    </span>
  </styled.span>
</CartCount>
`;
